### PR TITLE
Move files from os-common

### PR DESCRIPTION
**Pull request by chaitu236**
Move nisetreboottype, xfce files from os-common

@ni/rtos

### CODE REVIEW

File: recipes-xfce/xfce-nilrt-settings/xfce-nilrt-settings_1.0.bb
Comment (shruthi-ravi) [inline recipes-xfce/xfce-nilrt-settings/xfce-nilrt-settings_1.0.bb]: Here's another [recipe](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-xfce/xserver-xfce-init/xserver-xfce-init_1.0.bb#L27) that adds this file. I think you can merge the contents of this default file to that file. I think that would be the part where we read from the nirtcfg settings to check if UI is enabled.

File: recipes-ni/initscripts-nilrt/files/mountconfig
Comment (amstewart) [inline recipes-ni/initscripts-nilrt/files/mountconfig]: The `config` partition only exists on ARM targets, iirc. So I don't think this initscript is relevant to dunfell images and beyond. It can be dropped.

File: recipes-ni/initscripts-nilrt/files/nicreatelogdir
Comment (amstewart) [inline recipes-ni/initscripts-nilrt/files/nicreatelogdir]: Do we really need an initscript for this? Surely there is either (a) already another initscript which creates NI-owned directories or (b) we can do what every other distro does and expect that log-creating packages will make their own directories.

File: recipes-ni/initscripts-nilrt/files/nisetreboottype
Comment (amstewart) [inline recipes-ni/initscripts-nilrt/files/nisetreboottype:5]: I don't know if this is a relevant workflow; it seems like the remnants of the obsolete "install mode" workflow. Can you check if SCS or anyone else even sets `/sys/kernel/ni_requested_reboot_type` to `install`? If not, we can drop this whole file.

File: recipes-ni/initscripts-nilrt/files/mountconfig
Comment (shruthi-ravi) [inline recipes-ni/initscripts-nilrt/files/mountconfig]: The entries that refer to mtd and mount_ubi_volume are Zynq-specific and can be removed. The mount command on line 33 is enough. Do you think it's a good idea to edit these files as you move them or should it be done in the next update? Can you create a task for it or something if you pick the latter option?

File: recipes-ni/initscripts-nilrt/files/nisetupdump
Comment (amstewart) [inline recipes-ni/initscripts-nilrt/files/nisetupdump]: We need to keep these settings*, but the rest of this script is only relevant to safemode. Surely we already have a package in OE that sets `ulimit` or something? That's probably a better place to put sysctl options like this.

* Though the `core_pattern` should be set to a better location. Where should that be @gratian ?

File: recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb
Comment (shruthi-ravi) [inline recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb]: Description feels redundant.

File: recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb
Comment (amstewart) [inline recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb:94]: This script will eventually need to be on RL 4 as well, but we can do that in the later RAUC story to configure RL 4 as the new "safemode".

File: recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb
Comment (shruthi-ravi) [inline recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb]: This [recipe](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-support/syslog-ng/syslog-ng_3.%25.bbappend#L25) seems to be creating the same directory.

File: recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb
Comment (shruthi-ravi) [inline recipes-ni/initscripts-nilrt/initscripts-nilrt_1.0.bb]: I think it would be more appropriate to move this description to the script.

File: recipes-xfce/xfce-nilrt-settings/files/xserver-xfce.default
Comment (amstewart) [inline recipes-xfce/xfce-nilrt-settings/files/xserver-xfce.default]: Seems like this file already exists in [xserver-xfce-init-1.0](https://github.com/ni/meta-nilrt/blob/29fbbd49b52e84de6fb1b3c18702b743d2124420/recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf) and is installed to the same place [here](https://github.com/ni/meta-nilrt/blob/29fbbd49b52e84de6fb1b3c18702b743d2124420/recipes-xfce/xserver-xfce-init/xserver-xfce-init_1.0.bb#L27).

**Review by amstewart — CHANGES_REQUESTED** [review]

**Review by shruthi-ravi — CHANGES_REQUESTED** [review]
Part 1..I still have the last 3 commits to look at.

**Comment by shruthi-ravi** [comment]
Well, looks like I was a minute late :) @amstewart has similar concerns.

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (shruthi-ravi) [inline recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Should this really be an error? I don't understand why. This seems to be the only difference between existing code and new code, other than the comment itself?

File: recipes-ni/initscripts-nilrt/files/nicreatelogdir
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nicreatelogdir]: It does seem excessive to have an initscript just for this. But I [haven't been able to find](https://codesearch.natinst.com/search/text?q=mkdir.*natin%20repo%3Ameta-nilrt&fold_case=false&regex=true&context=true) another initscript that creates NI dirs.

As for (b), I suspect the reason this was added was to support some old piece of software that didn't creates its own dir. Do you think we should remove this to find all such cases and have their devs fix them?

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (amstewart) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Yeah; I agree that this doesn't imply an error.

File: recipes-ni/initscripts-nilrt/files/nicreatelogdir
Comment (amstewart) [reply recipes-ni/initscripts-nilrt/files/nicreatelogdir]: >Do you think we should remove this to find all such cases and have their devs fix them?

It [seems like](https://codesearch.natinst.com/search/text?q=local%2Fnatinst%2Flog&fold_case=auto&regex=true&context=true) that path is used in a lot of different packages. In all cases, it seems like the intention is just to give the `lvuser` user a location to access core dumps, LV crash logs, ect. (which implies that it is where LV customers are intended to store their logs).

It's probably too risky to just remove it from the base image and require the NIFeeds package owners to fix all their packages. But I might like to move this code to something like the [`niacctbase`](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-core/niacctbase/niacctbase.bb) package, which does generic setup for the `lvuser` account.

Comment (shruthi-ravi) [reply recipes-ni/initscripts-nilrt/files/nicreatelogdir]: I found a [recipe](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-support/syslog-ng/syslog-ng_3.%25.bbappend#L25) that creates the directory too.

File: recipes-ni/initscripts-nilrt/files/nisetupdump
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupdump]: There is [this](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-ni/base-files-nilrt/files/ulimit.sh#L1) but it gets installed to /etc/profile.d, which is probably not where we want to add this.
How about [nisetupkernelconfig](https://github.com/ni/meta-nilrt/blob/nilrt/master/sumo/recipes-ni/initscripts-nilrt/files/nisetupkernelconfig#L1)?

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Should I change the msg to "No IRQ threads found" and exit 0? [Ref](https://review-board.natinst.com/r/138441/)

Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: @billpittman What do you think?

Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Ok I just tried the original and don't see an error when there are no irq threads - so not sure why we added it.

File: recipes-ni/initscripts-nilrt/files/mountconfig
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/mountconfig]: Removed the file altogether as remaining changes aren't necessary. I can edit them as I go.

File: recipes-ni/initscripts-nilrt/files/nisetreboottype
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetreboottype:5]: Found [this](https://p4.natinst.com/browser/perforce/MAX/components/remote/softwareinstall/trunk/21.0/source/SoftwareActionShared/Imaging.cpp?revision_at=51501451#line_388)... so keep this around?

Comment (amstewart) [reply recipes-ni/initscripts-nilrt/files/nisetreboottype:5]: I... don't know what that's supposed to mean? Does that codepath run when `reboot_type` is set to `install`?

Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetreboottype:5]: The codepath runs when ni_boot_mode_input is set to install and [this](https://p4.natinst.com/browser/perforce/MAX/components/remote/softwareinstall/trunk/21.0/source/SoftwareActionShared/Imaging.cpp?revision_at=51501451#line_388) does that.

File: recipes-ni/initscripts-nilrt/files/nisetupdump
Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupdump]: Adding core_pattern also to nisetupkernelconfig for now

**Comment by amstewart** [comment]
@chaitu236 Can you post a V2 summary, so I know what you changed.

**Comment by chaitu236** [comment]
> @chaitu236 Can you post a V2 summary, so I know what you changed.

Addressed all feedback except for the one in nisetupkernelconfig.

Removed mountconfig, nicreatelogdir, xserver-xfce.default
Moved nisetupdump  stuff into nisetupkernelconfig

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (shruthi-ravi) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Ok, I think we can keep the original file with the updated comment unless @gratian objects.

File: recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf
Comment (shruthi-ravi) [inline recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf:6]: Does this have the effect of exiting from the parent xserver-xfce process too? I assume that is the intent given the message.

**Review by shruthi-ravi — COMMENTED** [review]
lgtm, pending the two outstanding issues.

File: recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf
Comment (gratian) [reply recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf:6]: The intent of the original code was to not start the X stack (X server and Xfce desktop environment) if the "Enable Embedded UI" checkbox in MAX is not checked.

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (gratian) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: I'll defer to @billpittman since he's the original author. My guess is the error is there to catch a case where this script is not working e.g. the kernel changed the names of irq threads or we misconfigured the kernel and interrupts are not threaded.

File: recipes-ni/initscripts-nilrt/files/nisetupdump
Comment (gratian) [reply recipes-ni/initscripts-nilrt/files/nisetupdump]: Yeah, I think moving the core_pattern  into nisetupkernelconfig is OK.

File: recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf
Comment (shruthi-ravi) [reply recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf:6]: Right. Since this file is invoked from [here](https://github.com/ni/meta-nilrt/blob/nilrt/master/dunfell/recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce#L8), I was wondering if this exit would cause the parent process to also finish or would it continue on to line 26, which does not seem to be the intent.

File: recipes-ni/initscripts-nilrt/files/nisetupkernelconfig
Comment (billpittman) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Definitely does not need to be an error and I think Gratian instincts about the error being there in the first place are correct.  You could probably make a case for removing it all together seeing as it will be corrected eventually and with this running at startup (and the suggested practice of warm-up iterations), there are multiple opportunities for the threads/IRQs to be scheduled/re-scheduled.

Comment (chaitu236) [reply recipes-ni/initscripts-nilrt/files/nisetupkernelconfig:32]: Changed error to warning and exit 0

**Comment by chaitu236** [comment]
v3 changes: Changed nisetupkernelconfig's error to warning and exit 0

**Review by amstewart — APPROVED** [review]

File: recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf
Comment (shruthi-ravi) [reply recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf:6]: Do you have an update on this @chaitu236?

Comment (chaitu236) [reply recipes-xfce/xserver-xfce-init/xserver-xfce-init-1.0/xserver-xfce.conf:6]: Oh sorry I misunderstood the comment. Yes the exit would cause the parent to also exit and it wouldn't continue to line 26.

**Review by shruthi-ravi — APPROVED** [review]

**Review by amstewart — APPROVED** [review]

**Pull request merged**
